### ISSUE TITLE
match_ref_pats: don't suggest if there's only a single & to be removed

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#5281]
The lint will suggest to change
````rust
        match self {
            &AlignedTestName(_, p) => p,
            _ => PadNone,
        }
````
to 
````rust
        match *self {
            AlignedTestName(_, p) => p,
            _ => PadNone,
        }
````
IMO that's not much of an improvement if there is only a single pattern that is changed.

**Comment by Luro02** [comment]
I like the latter more than the former version, because it is future proof. For example one might add a new variant to the match (without changing the self to *self):

```rust
match self {
    &AlignedTestName(_, p) => p,
    &Other(x) => unimplemented!(x),
    _ => PadNone,
}
```

instead of

```rust
match *self {
    AlignedTestName(_, p) => p,
    Other(x) => unimplemented!(x),
    _ => PadNone,
}
```

but I agree that this is quite a minor improvement.

**Comment by matthiaskrgr** [comment]
Perhaps the single-pattern case could be sourced out into a pedantic/allow-by-default lint so that it still warns on multiple &patterns or `match &self { &Pat => ....` as before.

**Comment by camsteffen** [comment]
The lint also covers `if let &..`. which is "one arm". I think the lint does what it's designed to do.
